enterprise/frontend/src/embedding-sdk/lib/theme/get-embedding-theme.ts

```diff
@@ -19,6 +20,12 @@ import { colorTuple } from "./color-tuple";
 
 const SDK_BASE_FONT_SIZE = `${DEFAULT_SDK_FONT_SIZE}px`;
 
+// Strip any key that has the value of "undefined"
+const stripUndefinedKeys = <T>(x: T): unknown =>
+  _.isObject(x)
+    ? _.pick(_.mapObject(x, stripUndefinedKeys), v => !_.isUndefined(v))
+    : x;
```

Comment:
If I define a theme object like `{components: {popover: {zIndex: undefined}}}` then the zIndex should fallback to the default theme values (e.g. 200), and not literally be `undefined`. Added the test case for this below.

If I instead pass `null` or other non-undefined values, it should still be applied.

Reply:
I don't know if between this or using the [associator](https://github.com/aearly/icepick?tab=readme-ov-file#mergetarget-source-associator), would be simpler. I think if we use the associator, it'd be a simple function that we'd ignore the nully value from `theme.components`

Reply:
@WiNloSt nice, I didn't knew that associator argument existed! let me update the code

Reply:
@WiNloSt I think icepick's latest type definition is still 2.3.3 which does not have a resolver yet, as that is available in 2.4.0. https://www.npmjs.com/package/@types/icepick seems to be outdated 😞 

```ts
export function merge<T, S1>(target: T, source: S1): T & S1;
```

I'll keep the existing code for now.

Reply:
well yeah. that sucks. Thanks for trying it out tho.